src/gui/optionsdlg.cpp

```diff
@@ -473,10 +612,63 @@ void OptionsDialog::saveOptions()
         qApp->installTranslator(translator);
     }
 
+    // Appearance preferences
+
+    GuiIconProvider::setIconSet(static_cast<GuiIconProvider::IconSet>(m_ui->comboIconTheme->currentIndex()));
+
+    const auto applyTheme = [this](Theme::Kind kind, const QString &themeName)
+    {
+
+        const auto themeKindToString = [](Theme::Kind kind)
+        {
+            switch (kind) {
+                case Theme::Kind::Color:
+                    return tr("color", "e.g. 'color theme'");
+                case Theme::Kind::Font:
+                    return tr("font", "e.g. 'font theme'");
+                default:
+                    return tr("unknown", "e.g. 'unknown theme'");
+            }
+        };
+
+        using namespace Theme::Serialization;
+        QString errorMessage;
+        try {
+            Theme::ThemeProvider::instance().setCurrentTheme(kind, themeName);
+            return;
+        }
+        catch (ThemeElementMissing &ex) {
+            errorMessage = tr("theme element '%1' is missing").arg(ex.elementName());
+        }
+        catch (ValueParsingError &ex) {
+            errorMessage = tr("could not parse element value '%1'").arg(ex.valueString());
+        }
+        catch (ParsingError &ex) {
+            errorMessage = tr("Could not parse element '%1'").arg(ex.serializedValue());
+        }
+        catch (UnknownProvider &ex) {
+            errorMessage = tr("theme element contains unknown scheme '%1'").arg(ex.profiderName());
```

Comment:
@evsh:  Typo alert: 

Should be `(ex.providerName())`, rather than `(ex.profiderName())`.